### ISSUE TITLE
Script I made was working before, now it shows Read Timeout error

### ISSUE THREAD

**Issue by rmusngi** [GeneralMills/pytrends#461]
Hi, I have been using a script to get Google Trends data but today it suddenly showed this error `HTTPSConnectionPool(host='trends.google.com', port=443): Read timed out. (read timeout=5)`

Is it a problem in my code or somewhere else?

**Comment by YohanDhana** [comment]
I'm also having the same problem : \

**Comment by QimanWang** [comment]
same here, likely google modified their api proxy policy

**Comment by hunpark19** [comment]
now, 429 error returned. sleeping is not working. seems that google made black-list ip. any idea ?

**Comment by burton030** [comment]
I'm having the exact same problem. First the time out error now the 429 error.

**Comment by anjadevries** [comment]
same here, has anyone been able to fix it?

**Comment by rmusngi** [comment]
I made the timeout 10,25 but it made the script run so slow :(

**Comment by burton030** [comment]
This does not work for me.

**Comment by tylercroberts** [comment]
Also have this same issue, sleeping longer has not done much. Has anyone had success in fixing this?

**Comment by iJohnMaged** [comment]
Same issue here. Are you all using AWS or something? I've run it once on my local PC and it ran smoothly but on aws I'm getting 429.

**Comment by hunpark19** [comment]
@iJohnMaged
 Yes. on AWS. I tested on 3 different regions, all the same.

**Comment by hunpark19** [comment]
I've always been wondering why google trends doesn't have official API like google maps, etc.

**Comment by burton030** [comment]
I tried it on two different databricks cluster. For the one, I'm usually using it does not work anymore for the other one it did.

**Comment by rlancer** [comment]
Experiencing the issue on AWS as well

**Comment by cmeier303** [comment]
Same issue here on my pc and on AWS.  AWS instance has been blocked for over 10 days.  2 other computers I tried started right from the get go with timeouts.

**Comment by Korpip** [comment]
I am getting the same error running this from my pizero from my home.

**Comment by silviacamplani** [comment]
Same problem here, it has been three days since I am getting error 429 on my EC2 instance. A greater sleep or timeout has not resolved the problem.

**Comment by philippschmalen** [comment]
I got the same problem and narrowed it down to the `related_queries()` function. Other functions seem to be working well. 
Hope to see a solution for it anytime soon!

**Comment by iJohnMaged** [comment]
> 
> 
> I got the same problem and narrowed it down to the `related_queries()` function. Other functions seem to be working well.
> Hope to see a solution for it anytime soon!

Nope. All others same issue.

**Comment by silviacamplani** [comment]
> Same issue here. Are you all using AWS or something? I've run it once on my local PC and it ran smoothly but on aws I'm getting 429.

@iJohnMaged  I'm just curious but did you made too many requests before getting 429?

**Comment by iJohnMaged** [comment]
> 
> 
> > Same issue here. Are you all using AWS or something? I've run it once on my local PC and it ran smoothly but on aws I'm getting 429.
> 
> @iJohnMaged I'm just curious but did you made too many requests before getting 429 on your AWS instance?

Nope. I started getting it on my 3rd request, locally I can get away with 2500 requests before getting any 429s and sleeping for 60s solves it.

**Comment by guareber** [comment]
Reporting similar issue from AWS - multiple regions tested in EU and US with new public IPs,  all to the same result: only 1-3 requests allowed per period, the rest end up with a 429 - local seems to be just like before.
It would appear google has decided to severely restrict AWS IP ranges from using the service.

**Comment by dev-dimsum** [comment]
Do you think it'll still be throttled to 3 requests/min if you move to GCP?

**Comment by iJohnMaged** [comment]
> 
> 
> Do you think it'll still be throttled to 3 requests/min if you move to GCP?

Working fine on GCP :)

**Comment by tyoc213** [comment]
@iJohnMaged  How many request can be send from GCP?

Could be that this limit is only applied to AWS??? 

I thought this was a different error than the 429 reported time a go https://github.com/GeneralMills/pytrends/issues/435

**Comment by iJohnMaged** [comment]
> 
> 
> @iJohnMaged How many request can be send from GCP?
> 
> Could be that this limit is only applied to AWS??? Is this 2-3 request per minute visible somewhere???? or is just known now by trrial and error (reported experience)?
> 
> I thought this was a different error than the 429 reported time a go #435
> 
> Also **How many time do I need to sleep** between request? 1 minute or 1 hour? or it varies?

I think I'm getting about 2.5~3K on GCP almost with 5 seconds + random delay.

On AWS it's completely blocked, from the first request, at least for me.

**Comment by AndyMcGowan** [comment]
Having similar issues - running on AWS and no luck but works sporadically on local. 

Anyone on aws having any luck?

**Comment by kalagee-anjaria** [comment]
Yes I am also facing the same issue. When run locally, it works fine. But on AWS server I keep getting the 429 error. So is this problem due to the AWS ?
